### PR TITLE
Feat/integration 3.8 translation in publications

### PR DESCRIPTION
**Pull request by asf-algoo**
This PR adds translation buttons in publications. We will need to open an issue to factorize the code handling translation buttons when we have time.

## Checkpoints

**For developers**

- [X] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [X] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [ ] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [X] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [X] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [ ] Manual, quality tests have been done

### CODE REVIEW

File: frontend/src/container/FeedItemWithPreview.jsx [hunk 1]
```diff
@@ -153,6 +177,93 @@ export class FeedItemWithPreview extends React.Component {
...
+    const commentTranslationState = this.getCommentTranslationState(commentId)
+    this.setState({
+      translationStateByCommentId: {
+        ...this.state.translationStateByCommentId,
+        [commentId]: {
+          ...commentTranslationState,
+          translationState: getDefaultTranslationState(this.props.system.config)
+        }
+      }
+    })
```
Comment (gbonaspetti) [inline frontend/src/container/FeedItemWithPreview.jsx:225]:
```suggestion
    this.setState(prev => ({
      translationStateByCommentId: {
        ...prev.translationStateByCommentId,
        [commentId]: {
          ...commentTranslationState,
          translationState: getDefaultTranslationState(this.props.system.config)
        }
      }
    }))
```

File: frontend_app_html-document/src/container/HtmlDocument.jsx [hunk 2]
```diff
@@ -843,31 +842,24 @@ export class HtmlDocument extends React.Component {
...
-    const translatedRawContent = await response.text()
-    this.setState({ translatedRawContent, translationState: TRANSLATION_STATE.TRANSLATED })
   }
 
   handleRestoreDocument = () => {
-    this.setState(previousState => {
-      return { translationState: getDefaultTranslationState(previousState.config.system.config) }
+    this.setState({
+      translationState: getDefaultTranslationState(this.state.config.system.config)
     })
```
Comment (gbonaspetti) [inline frontend_app_html-document/src/container/HtmlDocument.jsx:863]:
```suggestion
    this.setState(prev => ({
      translationState: getDefaultTranslationState(prev.config.system.config)
    }))
```

**Review by gbonaspetti — CHANGES_REQUESTED** [review]

File: frontend/src/container/FeedItemWithPreview.jsx [hunk 3]
```diff
@@ -220,13 +340,25 @@ export class FeedItemWithPreview extends React.Component {
...
                 workspaceId={props.workspaceId}
+                onClickTranslateComment={(
+                  comment => handleTranslateComment(
+                    FETCH_CONFIG.apiUrl,
+                    props.content.workspaceId,
+                    props.content.id,
+                    comment.content_id,
+                    props.i18n.language,
+                    props.system.config,
+                    this.commentSetState.bind(this, comment.content_id)
```
Comment (Skylsmoi) [inline frontend/src/container/FeedItemWithPreview.jsx:358]: is it required to bind this here?

**Review by gbonaspetti — APPROVED** [review]

File: frontend/src/container/FeedItemWithPreview.jsx (hunk 3 above)
Comment (raphj) [reply frontend/src/container/FeedItemWithPreview.jsx:358]: I am using bind to set the first comment, `this` is kind of irrelevant here but it is a necessary argument.

Comment (raphj) [reply frontend/src/container/FeedItemWithPreview.jsx:358]: I guess I can write this as:

```suggestion
                    () => this.commentSetState(comment.content_id)
```

**Review by Skylsmoi — APPROVED** [review]

**Pull request merged**